core/src/test/java/org/elasticsearch/common/xcontent/BaseXContentTestCase.java

```diff
@@ -19,23 +19,665 @@
+        assertThat(parser.nextToken(), is(Token.END_OBJECT));
+        assertNull(parser.nextToken());
+    }
+
+    public void testText() throws Exception {
+        assertResult(() -> builder().startObject().field("text", (Text) null).endObject(),
+                equalToJSon("{'text':null}"));
+        assertResult(() -> builder().startObject().field("text", new Text("")).endObject(),
+                equalToJSon("{'text':''}"));
+        assertResult(() -> builder().startObject().field("text", new Text("foo bar")).endObject(),
+                equalToJSon("{'text':'foo bar'}"));
+
+        final BytesReference random = new BytesArray(randomBytes());
+        XContentBuilder builder = builder().startObject().field("text", new Text(random)).endObject();
+
+        XContentParser parser = xcontentType().xContent().createParser(builder.bytes());
+        assertThat(parser.nextToken(), is(Token.START_OBJECT));
+        assertThat(parser.nextToken(), is(Token.FIELD_NAME));
+        assertThat(parser.currentName(), equalTo("text"));
+        assertThat(parser.nextToken().isValue(), is(true));
+        assertThat(parser.utf8Bytes().utf8ToString(), equalTo(random.utf8ToString()));
+        assertThat(parser.nextToken(), is(Token.END_OBJECT));
+        assertNull(parser.nextToken());
+    }
+
+    public void testReadableInstant() throws Exception {
+        assertResult(() -> builder().startObject().field("instant", (ReadableInstant) null).endObject(),
+                equalToJSon("{'instant':null}"));
+        assertResult(() -> builder().startObject().field("instant").value((ReadableInstant) null).endObject(),
+                equalToJSon("{'instant':null}"));
+
+        final DateTime t1 = new DateTime(2016, 1, 1, 0, 0, DateTimeZone.UTC);
+        assertResult(() -> builder().startObject().field("t1", t1).endObject(),
+                equalToJSon("{'t1':'2016-01-01T00:00:00.000Z'}"));
+        assertResult(() -> builder().startObject().field("t1").value(t1).endObject(),
+                equalToJSon("{'t1':'2016-01-01T00:00:00.000Z'}"));
+
+        final DateTime t2 = new DateTime(2016, 12, 25, 7, 59, 42, 213, DateTimeZone.UTC);
+        assertResult(() -> builder().startObject().field("t2", t2).endObject(),
+                equalToJSon("{'t2':'2016-12-25T07:59:42.213Z'}"));
+        assertResult(() -> builder().startObject().field("t2").value(t2).endObject(),
+                equalToJSon("{'t2':'2016-12-25T07:59:42.213Z'}"));
+
+        final DateTimeFormatter formatter = randomFrom(ISODateTimeFormat.basicDate(), ISODateTimeFormat.dateTimeNoMillis());
+        final DateTime t3 = DateTime.now();
+        assertResult(() -> builder().startObject().field("t3", t3, formatter).endObject(),
+                equalToJSon("{'t3':'" + formatter.print(t3) + "'}"));
+        assertResult(() -> builder().startObject().field("t3").value(t3, formatter).endObject(),
+                equalToJSon("{'t3':'" + formatter.print(t3) + "'}"));
+
```

Comment:
Is it worth adding a random test here? I'd love to have tests for far future and far past dates and that'd cover it.